### PR TITLE
v1 Metrics - Usage

### PR DESCRIPTION
**Pull request by gsvr**
- [x] Number of users Total - total and over time
- [x] Average USSD sessions per user needed to complete Clinic Registration
- [x] Average USSD sessions per user needed to complete CHW Registration
- [x] Average USSD sessions per user needed to complete public subscription

### CODE REVIEW

**Comment by gsvr** [comment]
@imsickofmaps Please review

Note: code is incomplete, pull-review sent for checking by maps before implementing in other channels.

**Comment by gsvr** [comment]
@imsickofmaps Please review.  I'd advise not looking at the go-app-... files when reviewing, otherwise you could do double reviewing.

File: go-app-personal.js [hunk 1]
```diff
@@ -127,12 +137,30 @@ go.app = function() {
...
+                ]);
+
+            });
 
             self.im.on('session:close', function(e) {
                 if (!self.should_send_dialback(e)) { return; }
                 return self.send_dialback();
             });
 
+            // can't get this event-handler to fire in the tests - no tests!
```
Comment (imsickofmaps) [inline go-app-personal.js]: @smn / @justinvdm / @hodgestar - any of you see what we're missing here to fire this event in tests?

File: go-app-chw.js [hunk 2]
```diff
@@ -175,6 +203,16 @@ go.app = function() {
                 });
         };
 
+        // unique users for the account (across conversations)
+        self.get_unique_users = function() {
```
Comment (imsickofmaps) [inline go-app-chw.js]: Called `get` but actually fires... rename please

**Comment by imsickofmaps** [comment]
@smn ready for review... but needs help as per https://github.com/praekelt/go-ndoh/pull/19#discussion_r12424669

File: src/chw.js [hunk 3]
```diff
@@ -13,12 +14,30 @@ go.app = function() {
...
+
+            });
 
             self.im.on('session:close', function(e) {
                 if (!self.should_send_dialback(e)) { return; }
                 return self.send_dialback();
             });
 
+            // can't get this event-handler to fire in the tests - no tests!
+            self.im.on('new:user', function(e) {
```
Comment (justinvdm) [inline src/chw.js]: s/`new:user`/`user:new`

Comment (justinvdm) [inline src/chw.js]: Ah sorry, only noticed now, we need to listen to `im.user`, so `self.im.user.on('user:new', function() { ... });`

**Comment by gsvr** [comment]
@imsickofmaps Please review latest commits

**Comment by imsickofmaps** [comment]
@smn issues addressed. Please review.

File: go-app-chw.js [hunk 4]
```diff
@@ -175,6 +202,16 @@ go.app = function() {
                 });
         };
 
+        // unique users for the account (across conversations)
+        self.get_and_fire_unique_users = function() {
+            return self.im
+                .api_request('messagestore.count_inbound_uniques',{})
```
Comment (smn) [inline go-app-chw.js]: This returns counts for the current conversation, not across all conversations. If you want to get it for a different conversation you need to request it per conversation key.

**Comment by smn** [comment]
:+1: looks good but make sure you're getting the info you're after with `messagestore.count_inbound_uniques`

**Pull request merged**